daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -270,22 +270,41 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
       ledgerTime: Time.Timestamp,
       submissionTime: Time.Timestamp,
       seeding: speedy.InitialSeeding,
-  )(implicit loggingContext: LoggingContext): Result[(SubmittedTransaction, Tx.Metadata)] =
+  )(implicit loggingContext: LoggingContext): Result[(SubmittedTransaction, Tx.Metadata)] = {
+    import Result.Implicits._
```

Comment:
@nickchapman-da @samuel-williams-da what do you think about refactoring the engine in this way ? 
Not completely, if it is a good idea, would like to get your thought. 
This basically a prerequisite for my submission script.  If you think it is too ugly, then I will just push it to a PoC branch, where I will develop the rest of the submission script implementation